src/main/java/org/mvel2/execution/ExecutionArrayList.java

```diff
@@ -116,13 +165,134 @@ public long memorySize() {
         return memorySize;
     }
 
+    public int indexOf(Object o, int fromIndex) {
+        int idx = this.slice(fromIndex).indexOf(o);
```

Comment:
Wrong implementation (see basic demo here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/indexOf ). We are interested only in:

```
public int indexOf(Object o);
```